File: 프로그래머스/lv0/120885. 이진수 더하기/이진수 더하기.py
def solution(bin1, bin2):
    answer = ""
    tmp = 0
    for i in str(int(bin1) + int(bin2))[::-1]:
        answer += str(int(i) + tmp - 2 * (int(i) + tmp >= 2))
        tmp = (int(i) + tmp >= 2)*1
    return str(tmp)*(tmp == 1) + answer[::-1]

File: 프로그래머스/lv0/120885. 이진수 더하기/이진수 더하기.java

import java.util.*;
class Solution {
    public String solution(String bin1, String bin2) {
        long answer = 0;
        long tmp = 0;
        long i = 0;
        long b1 = Long.parseLong(bin1);
        long b2 = Long.parseLong(bin2);
        long summation = b1 + b2;
        while (summation != 0){
            tmp = summation % 10;
            summation = summation / 10 + tmp / 2;
            tmp = (tmp % 2) * (long) Math.pow(10, i);
            answer += tmp;
            i++;
        }
        return String.valueOf(answer);
    }
}